core/src/main/java/com/google/errorprone/bugpatterns/URLEqualsHashCode.java

```diff
@@ -83,7 +84,12 @@ public boolean matches(Tree tree, VisitorState state) {
         return false;
       }
       Type type = ASTHelpers.getType(tree);
-      if (!ASTHelpers.isSubtype(type, sym.type, state)) {
+      try {
+        if (!ASTHelpers.isSubtype(type, sym.type, state)) {
+          return false;
+        }
+      } catch (CompletionFailure e) {
+        // Some symbols necessary for the subtype check are not on the classpath; bail out
```

Comment:
Copied this pattern from `FunctionalInterfaceClash` and `argumentselectiondefects/Parameter`. But since the bug checkers contain ~22 calls to `ASTHelpers#isSubtype`, perhaps a more general  approach is in order. (And though not fixed in this PR, I hit the same issue with `RestrictToEnforcer`'s call to `ASTHelpers.findSuperMethods`, which has ~17 callers.)

Perhaps rather than giving up, Error Prone could only log a warning for any checker that causes a `CompletionFailure`. In the worst case this causes a false negative. This behavior could be made configurable on a per-plugin basis, but it's not clear that's worth the trouble. Or maybe crashing is better for genuine `ERROR` checks? :thinking: 

If you think the currently approach is fine then I can open a similar PR for the completion failure in `RestrictToEnforcer` (didn't dig into that code yet).

Reply:
This is fine for now.  Please go ahead with a similar approach for `RestrictToEnforcer`.